src/Microsoft.AspNetCore.Sockets/HttpConnectionDispatcher.cs

```diff
@@ -284,6 +284,16 @@ private ConnectionState CreateConnection(HttpContext context)
             await _manager.DisposeAndRemoveAsync(state);
         }
 
+        private async Task ExecuteApplication(EndPoint endpoint, Connection connection)
+        {
+            // Jump onto the thread pool thread so blocking user code doesn't block the setup of the 
+            // connection and transport
+            await AwaitableThreadPool.Yield();
```

Comment:
Is this really clearer than using `Task.Run` or are you just trying to be clever? It seems like unnecessary cleverness hiding the actual code to me.